artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientProducerCreditManagerImpl.java

```diff
@@ -16,20 +16,29 @@
  */
 package org.apache.activemq.artemis.core.client.impl;
 
+import java.lang.invoke.MethodHandles;
 import java.util.Iterator;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.concurrent.Executor;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
 import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.server.ActiveMQScheduledComponent;
 import org.apache.activemq.artemis.spi.core.remoting.SessionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ClientProducerCreditManagerImpl implements ClientProducerCreditManager {
 
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
    public static final int MAX_UNREFERENCED_CREDITS_CACHE_SIZE = 1000;
 
+   private static final int MAX_UNREFERENCED_CREDITS_CACHE_TIME_CHECK = 500; // milliseconds
```

Comment:
I set this as half second just for the test... but I'm actually thinking this should be something with a short timeout like this.. as long as the cleanup works.